### PR TITLE
resolve TODO in mapToUnstructured and sliceToUnstructured

### PR DESCRIPTION
**Pull request by kidlj**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

resolve TODO in mapToUnstructured and sliceToUnstructured: the original author seemed to have made a typo, that here should be `dv.Set(sv)` rather than `sv.Set(sv)`. After the correction, the tests should pass.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/runtime/converter.go [hunk 1]
```diff
@@ -706,20 +706,18 @@ func mapToUnstructured(sv, dv reflect.Value) error {
 	if dt.Kind() == reflect.Interface && dv.NumMethod() == 0 {
 		if st.Key().Kind() == reflect.String {
 			switch st.Elem().Kind() {
-			// TODO It should be possible to reuse the slice for primitive types.
-			// However, it is panicing in the following form.
```
Comment (wojtek-t) [inline staging/src/k8s.io/apimachinery/pkg/runtime/converter.go:710]: Hmm - it says it was panicing in the past and you're just uncommenting it.
Can you describe what has changed that it works now?

Comment (kidlj) [reply staging/src/k8s.io/apimachinery/pkg/runtime/converter.go:710]: Sure, I changed `sv.Set(sv)` to `dv.Set(sv)`, I think this could be a typo.

Comment (wojtek-t) [reply staging/src/k8s.io/apimachinery/pkg/runtime/converter.go:710]: Heh...

**Comment by fedebongio** [comment]
thank you @wojtek-t for taking a look!
/triage accepted

File: staging/src/k8s.io/apimachinery/pkg/runtime/converter.go [hunk 2]
```diff
@@ -706,20 +706,18 @@ func mapToUnstructured(sv, dv reflect.Value) error {
...
-			// case reflect.String, reflect.Bool,
-			// 	reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64,
-			// 	reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
-			// 	sv.Set(sv)
-			// 	return nil
+			case reflect.String, reflect.Bool,
```
Comment (wojtek-t) [inline staging/src/k8s.io/apimachinery/pkg/runtime/converter.go:709]: The failures from tests are related:
```
    --- FAIL: TestValidateCustomResourceDefinition/embedded_metadata_defaults (0.07s)
panic: interface conversion: interface {} is map[string]string, not map[string]interface {} [recovered]
	panic: interface conversion: interface {} is map[string]string, not map[string]interface {}
```

**Comment by kidlj** [comment]
Many tests fail, here is the reason:

Because of the code change in func `sliceToUnstructured`, 

```
	if dt.Kind() == reflect.Interface && dv.NumMethod() == 0 {
		switch st.Elem().Kind() {
		case reflect.String, reflect.Bool,
			reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64,
			reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
			dv.Set(sv)
			return nil
		default:
			// We need to do a proper conversion.
			dv.Set(reflect.MakeSlice(reflect.SliceOf(dt), sv.Len(), sv.Cap()))
			dv = dv.Elem()
			dt = dv.Type()
		}
	}
```

`[]string` will be set as `[]string` in `Unstructured`. However, many code requires it to be set as `[]interface{}`, for example, in `NestedStringSlice `:

```
func NestedStringSlice(obj map[string]interface{}, fields ...string) ([]string, bool, error) {
	val, found, err := NestedFieldNoCopy(obj, fields...)
	if !found || err != nil {
		return nil, found, err
	}
	m, ok := val.([]interface{})  // Here requires it to be []interface{}
	if !ok {
		return nil, false, fmt.Errorf("%v accessor error: %v is of the type %T, expected []interface{}", jsonPath(fields), val, val)
	}
...

```

The same goes for `mapToUnstructured`, `map[string]string` should be set as `map[string]interface{}`, so this fix is a breaking change to present API, which I think should be avoided.

So, should we just delete the TODO in `mapToUnstructured` and `sliceToUnstructured`, since they are not applicable anymore.

File: staging/src/k8s.io/apimachinery/pkg/runtime/converter.go (hunk 2 above)
Comment (kidlj) [reply staging/src/k8s.io/apimachinery/pkg/runtime/converter.go:709]: Yes, just found that this TODO is not applicable anymore, see comment. Should I delete the TODO?

Comment (wojtek-t) [reply staging/src/k8s.io/apimachinery/pkg/runtime/converter.go:709]: Yeah - you can remove it.

**Pull request merged**